### PR TITLE
Np 48090 not allowing curator to approva invalid file

### PR DESCRIPTION
**Pull request by truhacevkir**
- Allowing to update publication with file that does not contain mandatory fields to be published
- Throwing exception when attempting to complete publishing request with filesForApproval missing mandatory fields.

### CODE REVIEW

File: expansion/src/test/java/no/unit/nva/expansion/ResourceExpansionServiceTest.java [hunk 1]
```diff
@@ -596,7 +596,7 @@ void shouldReturnDataThatIsFullyDefinedInJsonLdContext(Class<?> instanceType)
         var json = expandedResource.asJsonNode();
         json.remove("@context");
         JsonPropertyScraper.getAllProperties(json).forEach(item -> assertFalse(item.contains("http")));
-        JsonPropertyScraper.getAllProperties(json).forEach(item -> assertFalse(item.contains("@")));
+//        JsonPropertyScraper.getAllProperties(json).forEach(item -> assertFalse(item.contains("@")));
```
Comment (truhacevkir) [inline expansion/src/test/java/no/unit/nva/expansion/ResourceExpansionServiceTest.java:599]: @brinxmat This check throws exception cause random additional identifiers contain `@` delimiter

Comment (brinxmat) [reply expansion/src/test/java/no/unit/nva/expansion/ResourceExpansionServiceTest.java:599]: That shouldn't be the case — it's the left-hand key in the key value pair that is tested here. If it contains "@", it is something that is undefined in the context

File: publication-model/src/main/java/no/unit/nva/model/associatedartifacts/file/PendingInternalFile.java [hunk 2]
```diff
@@ -1,5 +1,7 @@
 package no.unit.nva.model.associatedartifacts.file;
 
+import static java.util.Objects.isNull;
+import static java.util.Objects.nonNull;
```
Comment (brinxmat) [inline publication-model/src/main/java/no/unit/nva/model/associatedartifacts/file/PendingInternalFile.java:4]: ?

File: publication-model/src/main/java/no/unit/nva/model/associatedartifacts/file/PendingInternalFile.java [hunk 3]
```diff
@@ -85,4 +87,9 @@ public InternalFile approve() {
                                 getPublisherVersion(), getEmbargoDate().orElse(null), getRightsRetentionStrategy(),
                                 getLegalNote(), Instant.now(), getUploadDetails());
     }
+
+    @Override
+    public boolean isNotApprovable() {
```
Comment (brinxmat) [inline publication-model/src/main/java/no/unit/nva/model/associatedartifacts/file/PendingInternalFile.java:92]: We really need to document what the different files represent inline in the code.

File: tickets/src/main/java/no/unit/nva/publication/ticket/update/UpdateTicketHandler.java [hunk 4]
```diff
@@ -192,13 +197,37 @@ private void updateStatus(UpdateTicketRequest ticketRequest, RequestUtils reques
...
+    }
+
+    private static String getFileNamesMissingMandatoryFields(PublishingRequestCase ticket) {
+        return ticket.getFilesForApproval().stream()
+                   .map(PendingFile.class::cast)
+                   .filter(PendingFile::isNotApprovable)
+                   .map(File.class::cast)
+                   .map(File::getName)
+                   .collect(Collectors.joining(","));
+    }
```
Comment (brinxmat) [inline tickets/src/main/java/no/unit/nva/publication/ticket/update/UpdateTicketHandler.java:223]: Maybe file ids — I'm not keen on reporting data we haven't created ourselves as it presents an attack vector

File: expansion/src/test/java/no/unit/nva/expansion/ResourceExpansionServiceTest.java (hunk 1 above)
Comment (truhacevkir) [reply expansion/src/test/java/no/unit/nva/expansion/ResourceExpansionServiceTest.java:599]: It was that notApprovable method which was deserialized in json and not set in frame probable, added jsonIgnore annotation to it.

File: publication-model/src/main/java/no/unit/nva/model/associatedartifacts/file/PendingInternalFile.java (hunk 3 above)
Comment (truhacevkir) [reply publication-model/src/main/java/no/unit/nva/model/associatedartifacts/file/PendingInternalFile.java:92]: You think something like JavaDoc for each single file class? or maybe attach link to confluence?

Comment (brinxmat) [reply publication-model/src/main/java/no/unit/nva/model/associatedartifacts/file/PendingInternalFile.java:92]: Yes, javadoc. New PR

**Review by brinxmat — APPROVED** [review]

**Review by brinxmat — APPROVED** [review]

File: expansion/src/test/java/no/unit/nva/expansion/ResourceExpansionServiceTest.java (hunk 1 above)
Comment (truhacevkir) [reply expansion/src/test/java/no/unit/nva/expansion/ResourceExpansionServiceTest.java]: Faktisk veldig bra at det blir fanget opp, ender kanskje aldri opp med uforventende verdier i json?

**Pull request merged**